app_flutter/lib/task_box.dart

```diff
@@ -323,58 +322,99 @@ class TaskOverlayContents extends StatelessWidget {
+                            style: Theme.of(context).textTheme.bodyText1,
+                          ),
+                          if (isDevicelab(task))
+                            Text(
+                              'Attempts: ${task.attempts}\n'
+                              'Run time: ${runDuration.inMinutes} minutes\n'
+                              'Queue time: ${queueDuration.inSeconds} seconds\n'
+                              'Flaky: ${task.isFlaky}',
+                              style: Theme.of(context).textTheme.bodyText2,
+                            )
+                          else
+                            Text(
+                              'Task was run outside of devicelab',
+                              style: Theme.of(context).textTheme.bodyText2,
+                            ),
+                          if (isDevicelab(task)) TaskAttemptSummary(task: task),
+                        ],
+                      ),
+                    ),
+                  ],
                 ),
-              ProgressButton(
-                defaultWidget: const Text('Log'),
-                progressWidget: const CircularProgressIndicator(),
-                width: 60,
-                height: 50,
-                onPressed: _viewLog,
-                animate: false,
               ),
-              if (isDevicelab(task))
-                ProgressButton(
-                  defaultWidget: const Text('Rerun'),
-                  progressWidget: const CircularProgressIndicator(),
-                  width: 70,
-                  height: 50,
-                  onPressed: _rerunTask,
-                  animate: false,
-                ),
+              Row(
+                mainAxisAlignment: MainAxisAlignment.end,
+                children: <Widget>[
+                  if (isDevicelab(task))
+                    RaisedButton(
+                      child: Text.rich(
+                        TextSpan(
+                          text: 'FIND ',
+                          children: <TextSpan>[
+                            TextSpan(
+                              text: task.reservedForAgentId,
+                              style: const TextStyle(fontStyle: FontStyle.italic),
+                            ),
+                          ],
+                        ),
+                      ),
+                      onPressed: () {
+                        // Close the current overlay
+                        closeCallback();
+
+                        // Open the agent dashboard
+                        Navigator.pushNamed(
+                          context,
+                          AgentDashboardPage.routeName,
+                          arguments: task.reservedForAgentId,
+                        );
+                      },
+                    ),
+                  Padding(
+                    padding: const EdgeInsets.only(left: 8.0),
+                    child: ProgressButton(
+                      child: const Text('DOWNLOAD LOG'),
```

Comment:
nit: There's some confusion to this button for some users. Could you update this to `DOWNLOAD ALL LOGS`?

Some users thought this only downloaded the latest log.